pytorch_lightning/callbacks/device_stats_monitor.py

```diff
@@ -54,11 +55,15 @@ def on_train_batch_start(
         batch_idx: int,
         unused: Optional[int] = 0,
     ) -> None:
+        if not trainer.logger:
+            raise MisconfigurationException("Cannot use DeviceStatsMonitor callback with Trainer that has no logger.")
+
         if not trainer.logger_connector.should_update_logs:
             return
 
         device_stats = trainer.accelerator.get_device_stats(pl_module.device)
-        prefixed_device_stats = prefix_metrics_keys(device_stats, "on_train_batch_start")
+        separator = trainer.logger.group_separator
```

Comment:
what happens here in the case of a logger collection where loggers might have different separators?
Does the logger collection define this at all?

Reply:
It doesn't. But I think this is a major gap with the idea of logger collections itself, which is why I filed this issue to simplify this further: